hub/core/compression.py

```diff
@@ -764,130 +764,33 @@ def _read_audio_shape(
     return (info.num_frames, info.nchannels)
 
 
-def _strip_hub_mp4_header(buffer: bytes):
-    if buffer[: len(_HUB_MKV_HEADER)] == _HUB_MKV_HEADER:
-        return memoryview(buffer)[len(_HUB_MKV_HEADER) + 6 :]
-    return buffer
+def _decompress_video(file):
+    # int decompressVideo(unsigned char *file, int size, int ioBufferSize, unsigned char *decompressed, int isBytes, int nbytes)
+    # isBytes should be set to 1 in case of in-memory video else set to 0
+    # if isBytes is 1, size of file and internal buffer size must be set
+    # buffer size currently set to size of compressed data
 
-
-def _decompress_video(
-    file: Union[bytes, memoryview, str], nframes: Optional[int] = None
-) -> np.ndarray:
+    from hub.core.pyffmpeg._pyffmpeg import lib, ffi  # type: ignore
 
     shape = _read_video_shape(file)
-    command = [
-        ffmpeg_binary(),
-        "-i",
-        "pipe:",
-        "-f",
-        "image2pipe",
-        "-pix_fmt",
-        "rgb24",
-        "-vcodec",
-        "rawvideo",
-        "-",
-    ]
-    if isinstance(file, str):
-        command[2] = file
-        pipe = sp.Popen(command, stdout=sp.PIPE, stderr=sp.PIPE, bufsize=10 ** 8)
-        raw_video = pipe.communicate()[0]
-    else:
-        file = _strip_hub_mp4_header(file)
-        pipe = sp.Popen(
-            command, stdin=sp.PIPE, stdout=sp.PIPE, stderr=sp.PIPE, bufsize=10 ** 8
-        )
-        raw_video = pipe.communicate(input=file)[0]  # type: ignore
-    nbytes = len(raw_video)
-    if nframes is not None:
-        shape = (nframes,) + shape[1:]
-    size = np.prod(shape)
-    if nbytes >= size:  # size is computed from fps and duration, might not be accurate.
-        return np.frombuffer(memoryview(raw_video)[:size], dtype=np.uint8).reshape(
-            shape
-        )
-    else:  # If size was overestimated, append blank frames to the end.
-        arr = np.zeros(shape, dtype=np.uint8)
-        arr.reshape(-1)[: len(raw_video)] = np.frombuffer(raw_video, dtype=np.uint8)
-        return arr
-
-
-def _read_video_shape(file: Union[bytes, memoryview, str]) -> Tuple[int, ...]:
-    info = _get_video_info(file)
-    if info["duration"] is None:
-        nframes = -1
-    else:
-        nframes = math.floor(info["duration"] * info["rate"])
-    return (nframes, info["height"], info["width"], 3)
-
-
-def _get_video_info(file: Union[bytes, memoryview, str]) -> dict:
-    duration = None
-    command = [
-        ffprobe_binary(),
-        "-select_streams",
-        "v:0",
-        "-show_entries",
-        "stream=width,height,duration,r_frame_rate",
-        "-of",
-        "default=noprint_wrappers=1",
-        "pipe:",
-    ]
+    nbytes = np.prod(shape)
+    decompressed = ffi.new(f"unsigned char[{nbytes}]")
```

Comment:
Does this always holds ?

Reply:
Where it is de-allocated ?

Reply:
It holds for the uint8 videos that we are supporting

Reply:
Deallocation is taken care of by CFFI